views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,521 @@
+                    }else{
+                        this.changeDeviceFrame(element.value, true);
+                        this.scaleFrame();
+                        this.setupScreenSize();
+                        if (element.value === 'mobile'){
+                            self.controls.$mobileDevices.children('.mobile-device-selector').trigger('change');
+                        } else if( element.value === 'desktop'){
+                            self.controls.$desktopDevices.trigger('change');
+                        }
+        
+                    }
+                }
+        
+            };
+
+            this.controls = {
+                $deviceTypes: $(previewTypesTpl({
+                    items: deviceTypes,
+                })),
+                $mobileDevices: $(mobileDevicesTpl({
+                    items: devices.byType('mobile'),
+                })),
+                $desktopDevices: $(desktopDevicesTpl({
+                    items: devices.byType('desktop'),
+                }))
+            };
+
+
+            /**
+             *  when user changes device type he/she want to test item in
+             *  @event scale#preview-scale-device-type
+             */
+            this.controls.$deviceTypes.on('change', function (event) {
+               self.api.onDeviceTypeChange(event);
+               self.trigger('preview-scale-device-type', event.target.value);
+
+            });
+
+            /**
+             *  when user changes mobile device model he/she want to test item in
+             *  @event scale#preview-scale-device-mobile-type
+             */
+            this.controls.$mobileDevices.children('.mobile-device-selector').on('change', function (event) {
+                self.api.onDeviceChange(event);
+                self.trigger('preview-scale-device-mobile-type', event.target.value);
+
+            });
+
+            /**
+             *  when user changes mobile device screen orientation user want to test item in
+             *  @event scale#preview-scale-device-mobile-orientation-type
+             */
+            this.controls.$mobileDevices.children('.mobile-orientation-selector').on('change', function (event) {
+                self.api.onOrientationChange(event);
+                self.trigger('preview-scale-device-mobile-orientation-type', event.target.value);
+            });
+
+            /**
+             *  when user changes mobile device model he/she want to test item in
+             *  @event scale#preview-scale-device-desktop-type
+             */
+            this.controls.$desktopDevices.on('change', function (event) {
+                self.api.onDeviceChange(event);
+                self.trigger('preview-scale-device-desktop-type', event.target.value);
+            });
+
+            /**
+             * adjust device frame position and size when browser size change
+             */
+            $(window).on('resize orientationchange', function () {
```

Comment:
please namespace those events and detach them when the plugin destroys (idempotency principle)

Reply:
I am not able to namespace them because that are not custom events, it is native browser events, one for checking window resizing (https://developer.mozilla.org/en-US/docs/Web/API/Document/defaultView/resize_event) one for  mobile device orientation (https://developer.mozilla.org/en-US/docs/Web/API/Window/orientationchange_event)

Reply:
Detaching the events is still missing

Reply:
fixed

Reply:
Even if the events are not custom, since jQuery is used you can namespace them.
But that is not the bigger issue IMO. `resize` should never be used without proper control. This is an event that is triggered so often than it could simply freeze the browser if misused. So you should at least apply a `_.throttle` or a `_.debounce` on it.